ddtrace/tracer/option.go

```diff
@@ -534,6 +564,17 @@ func ChildOf(ctx ddtrace.SpanContext) StartSpanOption {
 	}
 }
 
+// WithContext associates the ctx with the span. This does not extract the
+// parent span from ctx, use StartSpanFromContext for this instead.
+// @TODO(REVIEWERS) We could make this private, but users could still modify
```

Comment:
Users will not be able to modify the config, even if exported, because `StartSpan*` does not allow passing a config, only functional options. I think you're right and this should be private.

Reply:
Yeah, private is fine. I'm just a bit bothered by the `Context` public field which I'm not sure how to make sense of in the API. I wonder if there is a better way... I guess the current architecture and layout is not ideal for this new requirement.